src/main/java/net/minecraftforge/common/extensions/ForgeRecipeManager.java

```diff
@@ -48,20 +54,42 @@ public void onResourceManagerReload(IResourceManager resourceManager) {
         CraftingHelper.reloadConstants(resourceManager);
     }
 
+    /**
+     * Returns all the recipes that match this type.
+     * @param type A recipe type.
+     * @return A mutable view of all the recipes that match this type.  Callers should not modify the returned list.
```

Comment:
`mutable` and `should not modify` are contradictory.
Either just remove mutable or say `immutable but live view`